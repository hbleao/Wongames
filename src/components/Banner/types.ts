export type ImageProps = {
  src: string;
};

export type BannerProps = {
  image: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
};
