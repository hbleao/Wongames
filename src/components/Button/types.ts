import React, {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  ReactNode
} from 'react';

type Size = 'small' | 'medium' | 'large';
type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = {
  size?: Size;
  fullWidth?: boolean;
  icon?: ReactNode;
  as?: React.ElementType;
} & ButtonTypes;

export type WrapperProps = {
  fullWidth?: boolean;
  size?: Size;
  hasIcon: boolean;
};
