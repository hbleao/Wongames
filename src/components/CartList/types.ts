import { GameItemProps } from 'components/GameItem/types';

export type CartListProps = {
  items: GameItemProps[];
  total: string;
};
