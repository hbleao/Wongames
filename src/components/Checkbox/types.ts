import { InputHTMLAttributes } from 'react';

export type CheckboxProps = {
  labelText?: string;
  labelFor?: string;
  isChecked?: boolean;
  labelColor?: 'black' | 'white';
  value?: string | ReadonlyArray<string> | number;
  onCheck?: (status: boolean) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = {
  labelColor?: 'black' | 'white';
};
