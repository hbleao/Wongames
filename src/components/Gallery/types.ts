export type GalleryImageProps = {
  src: string;
  alt: string;
  label: string;
};

export type GalleryProps = {
  items: GalleryImageProps[];
};

export type ModalProps = {
  isOpen: boolean;
};
