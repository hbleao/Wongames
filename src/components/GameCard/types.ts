import { RibbonColors, RibbonSizes } from 'components/Ribbon/types';

export type GameCardProps = {
  title: string;
  developer: string;
  image: string;
  price: string;
  promotionalPrice?: string;
  favorite?: boolean;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
  onFav: () => void;
};

export type PriceProps = {
  isPromotional?: boolean;
};
