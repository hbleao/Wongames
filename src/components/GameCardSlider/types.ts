import { GameCardProps } from 'components/GameCard/types';

export type GameCardSliderProps = {
  items: GameCardProps[];
  color?: 'white' | 'black';
};

export type WrapperProps = {
  color?: 'white' | 'black';
};
