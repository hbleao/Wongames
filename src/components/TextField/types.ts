import { InputHTMLAttributes, ReactNode } from 'react';

export type TextFieldProps = {
  labelText?: string;
  labelFor?: string;
  icon?: ReactNode;
  iconPosition?: 'left' | 'right';
  initialValue?: string;
  errorMessage?: string;
  disabled?: boolean;
  onInput?: (value: string) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type WrapperProps = Pick<TextFieldProps, 'disabled'> & {
  isError: boolean;
};

export type IconProps = Pick<TextFieldProps, 'iconPosition'>;
