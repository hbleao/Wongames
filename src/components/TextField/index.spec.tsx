import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TextField } from '.';

import { renderWithTheme } from 'utils/tests/renderWithTheme';

import { TextFieldProps } from './types';
import { Email } from '@styled-icons/material-outlined';

const makeSut = ({
  labelText,
  labelFor,
  initialValue,
  icon,
  iconPosition,
  disabled,
  errorMessage,
  onInput,
  ...props
}: TextFieldProps) => {
  const sut = renderWithTheme(
    <TextField
      labelText={labelText}
      labelFor={labelFor}
      initialValue={initialValue}
      icon={icon}
      iconPosition={iconPosition}
      disabled={disabled}
      errorMessage={errorMessage}
      onInput={onInput}
      {...props}
    />
  );

  return {
    sut
  };
};

describe('Components/TextField', () => {
  it('should renders without label', () => {
    const { sut } = makeSut({});

    const label = document.querySelector('label');

    expect(label).not.toBeInTheDocument();
    expect(sut.container).toMatchSnapshot();
  });

  it('should renders with placeholder', () => {
    const placeholder = 'jhon.doe@gmai.com';
    makeSut({ placeholder });

    const input = screen.getByPlaceholderText(placeholder);

    expect(input).toBeInTheDocument();
  });

  it('should renders with icon', () => {
    makeSut({ icon: <Email /> });

    const icon = screen.getByTestId('input-icon');

    expect(icon).toBeInTheDocument();
    expect(icon).toHaveStyle({
      order: 0
    });
  });

  it('should renders with icon on the left side', () => {
    makeSut({ icon: <Email />, iconPosition: 'left' });

    const icon = screen.getByTestId('input-icon');

    expect(icon).toHaveStyle({
      order: 1
    });
  });

  it('should changes its value when typing', async () => {
    const onInput = jest.fn();
    const text = 'any text';
    makeSut({ onInput });

    const input = screen.getByRole('textbox');

    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).toHaveValue(text);
      expect(onInput).toHaveBeenCalledTimes(text.length);
    });

    expect(onInput).toHaveBeenCalledWith(text);
  });

  it('should not be able changes its value when disabled', async () => {
    const onInput = jest.fn();
    const text = 'any text';
    makeSut({ onInput, disabled: true });

    const input = screen.getByRole('textbox');

    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).not.toHaveValue(text);
      expect(input).toBeDisabled();
    });

    expect(onInput).not.toHaveBeenCalledWith(text);
  });

  it('should be able render with error', async () => {
    const errorMessage = 'Something wrong';
    const labelText = 'Name';
    makeSut({
      labelText,
      labelFor: labelText,
      icon: <Email data-testid="icon" />,
      iconPosition: 'left',
      errorMessage
    });

    const errorMessageExists = screen.getByText(errorMessage);

    expect(errorMessageExists).toBeInTheDocument();
  });

  it('should is accessible by tab', async () => {
    makeSut({
      labelText: 'TextField',
      labelFor: 'TextField',
      id: 'TextField'
    });

    const body = document.body;
    expect(body).toHaveFocus();

    userEvent.tab();

    const input = screen.getByRole('textbox');

    await waitFor(() => {
      expect(input).toHaveFocus();
    });
  });
});
