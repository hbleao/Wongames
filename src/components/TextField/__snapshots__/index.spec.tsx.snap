// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/TextField should renders without label 1`] = `
.c0 {
  border-radius: 0.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c1 {
  background-color: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0.8rem 1.6rem;
  border: 0.2rem solid #EAEAEA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.8rem;
}

.c1:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c2 {
  width: 100%;
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  border: 0;
  background-color: transparent;
  outline: none;
  box-shadow: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        class="c2"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
